typescript/ws-protocol-examples/src/examples/mcap-record.ts

```diff
@@ -205,12 +216,20 @@ async function main(
             )) as McapChannelId;
             wsChannelsByMcapChannel.set(mcapChannelId, channel.id as WsChannelId);
 
-            log("subscribing to %s", channel.topic);
+            log("subscribing to %s (channel %d)", channel.topic, channel.id);
             const subscriptionId = client.subscribe(channel.id) as SubscriptionId;
             subscriptionsById.set(subscriptionId, { messageCount: 0, mcapChannelId });
           }),
         );
       });
+
+      client.on("unadvertise", (channelIds) => {
```

Comment:
this makes me wonder what happens if the websocket server unadvertises and then re-advertises the same channel id again? I forget if that is allowed in the protocol? If so I wonder if we would technically need to update `wsChannelsByMcapChannel` as well (or maybe that will cause more problems)